src/sentry/search/snuba/executors.py

```diff
@@ -1659,8 +1659,19 @@ def query(
         # Check if any search filters are in POSTGRES_ONLY_SEARCH_FIELDS
         search_filters = search_filters or ()
         group_ids_to_pass_to_snuba = None
+        too_many_candidates = False
         if any(sf.key.name in POSTGRES_ONLY_SEARCH_FIELDS for sf in search_filters):
-            group_ids_to_pass_to_snuba = list(group_queryset.values_list("id", flat=True))
+            max_candidates = options.get("snuba.search.max-pre-snuba-candidates")
+            group_ids_to_pass_to_snuba = list(
+                group_queryset.using_replica().values_list("id", flat=True)[: max_candidates + 1]
+            )
+        if (
+            group_ids_to_pass_to_snuba is not None
+            and len(group_ids_to_pass_to_snuba) > max_candidates
+        ):
+            metrics.incr("snuba.search.too_many_candidates", skip_internal=False)
+            too_many_candidates = True
+            group_ids_to_pass_to_snuba = []
```

Comment:
```suggestion
            if (too_many_candidates := (len(group_ids_to_pass_to_snuba) > max_candidates)):
              metrics.incr("snuba.search.too_many_candidates", skip_internal=False)
              group_ids_to_pass_to_snuba = None
```

You can keep this within same if that starts in line 1663, because that's the only way you'll have any candidates. 

Reseting to None when there are too many candidates will simplify some checks below.